activejob/test/jobs/enqueue_error_job.rb

```diff
@@ -3,17 +3,25 @@
 class EnqueueErrorJob < ActiveJob::Base
   class EnqueueErrorAdapter
     class << self
-      def enqueue(*)
-        raise ActiveJob::EnqueueError, "There was an error enqueuing the job"
-      end
+      attr_accessor :should_raise_sequence
+    end
+    self.should_raise_sequence = []
 
-      def enqueue_at(*)
-        raise ActiveJob::EnqueueError, "There was an error enqueuing the job"
-      end
+    def enqueue(*)
+      raise ActiveJob::EnqueueError, "There was an error enqueuing the job" if should_raise?
     end
+
+    def enqueue_at(*)
+      raise ActiveJob::EnqueueError, "There was an error enqueuing the job" if should_raise?
+    end
+
+    private
+      def should_raise?
+        self.class.should_raise_sequence.empty? || self.class.should_raise_sequence.shift
+      end
   end
 
-  self.queue_adapter = EnqueueErrorAdapter
+  self.queue_adapter = EnqueueErrorAdapter.new
```

Comment:
There was a need to convert this adapter to object based to get correct adapter name in the newly added tests.